### PR TITLE
ISSUE-603 [Public agenda] rest api to GET available slots (Public)

### LINKED ISSUE #603: [Public agenda] rest api to get booking info and available slots (Public)

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/twake-calendar-side-service/issues/603

## Summary by CodeRabbit

* **New Features**
  * Added a GET endpoint to fetch available slots for booking links (ISO-8601 range, max 60 days), returning JSON slots that exclude the link owner's busy times and provide detailed errors for invalid requests.

* **Tests**
  * New integration tests cover slot computation, edge cases, auth-free access, and error responses; DAO contract tests for public-id lookup.

* **Chores**
  * Registered the new route and an in-memory booking-link DAO for runtime/testing.
* **API**
  * Added public lookup by booking-link ID and a convenience constructor; improved not-found error message.

### CODE REVIEW

File: app/src/test/java/com/linagora/calendar/app/restapi/routes/BookingLinkSlotsRouteTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        assertThatJson(response)
+            .isEqualTo("""
+                {
+                  "durationMinutes": 30,
+                  "range": {
+                    "from": "2026-01-26T00:00:00Z",
+                    "to": "2026-01-27T00:00:00Z"
+                  },
+                  "slots": [
+                    "2026-01-26T09:00:00Z",
```
Comment (chibenwa) [inline app/src/test/java/com/linagora/calendar/app/restapi/routes/BookingLinkSlotsRouteTest.java]: `{"start": "2026-01-26T09:00:00Z"}`

Rationals: 
 - better description
 - ability to add more data in the future easily if needed (tokens to create events for instance?)

File: calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/BookingLinkSlotsRoute.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    private Instant parseRequiredInstant(QueryStringDecoder queryStringDecoder, String parameterName) {
+        try {
+            return queryStringDecoder.parameters().getOrDefault(parameterName, List.of())
+                .stream()
+                .findFirst()
+                .map(Instant::parse)
+                .orElseThrow();
+        } catch (RuntimeException e) {
+            throw new IllegalArgumentException("Missing or invalid query parameter '%s'. Expected ISO-8601 instant format, e.g. 2007-12-03T10:15:30.00Z"
+                .formatted(parameterName), e);
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/BookingLinkSlotsRoute.java]: endpoint contract changed, 
it support hh mm ss

**Review by hungphan227 — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
Overwize +1

File: app/src/test/java/com/linagora/calendar/app/restapi/routes/BookingLinkSlotsRouteTest.java [hunk 3]
```diff
@@ -0,0 +1,736 @@
...
+        davTestHelper = new DavTestHelper(sabreDavExtension.dockerSabreDavSetup().davConfiguration(), TECHNICAL_TOKEN_SERVICE_TESTING);
+    }
+
+    @Test
+    void shouldReturnSlotsWhenRequestIsValid(TwakeCalendarGuiceServer server) {
+        BookingLinkInsertRequest insertRequest = new BookingLinkInsertRequest(CalendarURL.from(openPaaSUser.id()), DURATION_30_MINUTES, AVAILABILITY_RULE);
+        BookingLink inserted = server.getProbe(BookingLinkSlotsProbe.class).insert(openPaaSUser.username(), insertRequest);
+
+        String response = given()
+            .pathParam("bookingLinkPublicId", inserted.publicId().value())
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply app/src/test/java/com/linagora/calendar/app/restapi/routes/BookingLinkSlotsRouteTest.java]: Example
`api/booking-links/384e5159-5663-45eb-a02f-626d34d6297d/`

It is generated by `UUID.randomUUID`
